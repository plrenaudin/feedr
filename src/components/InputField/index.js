import { linkEvent } from "inferno";

const handleChange = (props, event) => {
  props.onChange(event.target.value);
  reset();
};

const reset = () => (event.target.value = "");

const InputField = props => <input type="text" onChange={linkEvent(props, handleChange)} />;

export default InputField;
