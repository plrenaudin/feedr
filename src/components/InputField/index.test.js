import { render } from "inferno";
import InputField from ".";

it("Matches the snapshot", () => {
  const div = document.createElement("div");
  expect(render(<InputField placeholder={"test value"} />, div).dom).toMatchSnapshot();
});

it("Matches the snapshot when disabled", () => {
  const div = document.createElement("div");
  expect(render(<InputField placeholder={"test value"} disabled={true} />, div).dom).toMatchSnapshot();
});
