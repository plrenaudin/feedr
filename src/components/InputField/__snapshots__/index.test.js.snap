// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshot 1`] = `
<input
  placeholder="test value"
  type="text"
/>
`;

exports[`Matches the snapshot when disabled 1`] = `
<input
  disabled=""
  placeholder="test value"
  type="text"
/>
`;
